### PR TITLE
[Dataspace] - New Table Spec for BCR Data

### PR DESCRIPTION
**Pull request by labkey-bpatel**
#### Rationale
Dataspace client will be getting data about the B-Cell Receptors (BCRs). These BCR data are key in further understanding of HIV infections and possibly bringing us one step closer to the cure/vaccine. Once the data is in LabKey, the client will be writing R scripts to manipulate the data using LabKey APIs.

Spec [here](https://docs.google.com/document/d/1zM6BVb5y4bGVJXDzMCV8N5jTDqe6QABwxE7zd1sq57I/edit#)

#### Changes
* Create new tables to store BCR data (note: I have not added all the FKs mentioned in the spec since the data provided seems incomplete causing FK constraint errors during import).
* Add table metadata
* Bump module version
* Add sample data
* Add support to import csv files

### CODE REVIEW

File: resources/schemas/dbscripts/postgresql/cds-23.001-23.002.sql [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    container ENTITYID NOT NULL,
+
+    CONSTRAINT PK_cds_run_log PRIMARY KEY (run_number)
+);
+
+CREATE TABLE cds.allele_sequence
+(
+    allele VARCHAR(100) NOT NULL,
+    allele_sequence_nt VARCHAR(1000),
+    release_date DATE,
```
Comment (labkey-klum) [inline resources/schemas/dbscripts/postgresql/cds-23.001-23.002.sql]: should this be `TIMESTAMP`?

File: resources/schemas/dbscripts/postgresql/cds-23.001-23.002.sql [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    CONSTRAINT PK_cds_sequence_germline PRIMARY KEY (sequence_germline_id),
+    CONSTRAINT FK_cds_sequence_germline_sequence_id FOREIGN KEY (sequence_id) REFERENCES cds.sequence(sequence_id)
+);
+
+CREATE INDEX IX_cds_sequence_germline_sequence_id ON cds.sequence_germline(sequence_id);
+
+CREATE TABLE cds.antibody_sequence
+(
+    mab_id VARCHAR(100),
+    chain VARCHAR(10),
```
Comment (labkey-klum) [inline resources/schemas/dbscripts/postgresql/cds-23.001-23.002.sql]: `mab_id` and `chain` should be required

File: src/org/labkey/cds/data/CDSImportCopyConfig.java [hunk 3]
```diff
@@ -34,68 +34,88 @@
...
-public class CDSImportCopyConfig extends CopyConfig
+public abstract class CDSImportCopyConfig extends CopyConfig
 {
-    QueryUpdateService.InsertOption option = QueryUpdateService.InsertOption.IMPORT;
+    public abstract String getFileExtension();
+    public abstract DataIteratorBuilder getTabLoader(File file) throws IOException;
 
-    CDSImportCopyConfig(String sourceSchema, String source, String targetSchema, String target)
+    private String _fileName;
+    QueryUpdateService.InsertOption _option = QueryUpdateService.InsertOption.IMPORT;
```
Comment (labkey-klum) [inline src/org/labkey/cds/data/CDSImportCopyConfig.java:49]: `private final`?

File: src/org/labkey/cds/data/steps/ImportTask.java [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            if (null != source)
+            {
+                if (null == dir)
+                {
+                    logger.info("Copying data from " + config.getSourceSchema() + "." + config.getSourceQuery() + " to " +
+                            config.getTargetSchema() + "." + config.getTargetQuery());
+                }
+                else
+                {
+                    logger.info("Copying data from " + dir.toString() + " to " +
```
Comment (labkey-klum) [inline src/org/labkey/cds/data/steps/ImportTask.java]:
```suggestion
                    logger.info("Copying data from " + dir + " to " +
```

File: test/sampledata/dataspace/bcrimport/run_log.csv [hunk 5]
```diff
@@ -0,0 +1,3 @@
+run_number,run_application,run_information
+20230202_igblast,,
+20230202_vquest,,
```
Comment (labkey-klum) [inline test/sampledata/dataspace/bcrimport/run_log.csv:3]: do we anticipate more test data, or is that plan to wait for this to bake before we need it for automation etc.

**Review by labkey-klum — APPROVED** [review]

File: test/sampledata/dataspace/bcrimport/run_log.csv (hunk 5 above)
Comment (labkey-bpatel) [reply test/sampledata/dataspace/bcrimport/run_log.csv:3]: Yes, i had added sample files in my first commit, then unintentionally committed the delete of the files as i was testing something locally. Now they are added again.

**Pull request merged**